infra/postdeployment.tf

```diff
@@ -42,9 +42,10 @@ resource "google_compute_instance" "initialize" {
     google_cloud_run_v2_job.client,
   ]
 
-  name         = "head-start-initialize"
-  machine_type = "n1-standard-1"
-  zone         = var.zone
+  name           = "head-start-initialize"
+  machine_type   = "n1-standard-1"
+  zone           = var.zone
+  desired_status = "RUNNING"
```

Comment:
question: Confirming my understanding, without this `desired_status`, terraform would look, see we have an instance, and no-op. However, with this config in place, terraform is instructed to start up the GCE instance if it's not running.

Reply:
Correct. 

Terraform takes a convergence style approach: it does the minimum set of steps required to bring the infra back into the state documented. For the aspects of the GCE state described, that would reflect reality, so no changes would be made. Now that we're adding "Oh it also needs to be running", when we're elsewhere making sure it shuts down, TF will always assert that state, which has the side-benefit of running the metadata script (see later comment for why).